librz/il/definitions/effect.c

```diff
@@ -0,0 +1,213 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il/definitions/effect.h>
+
+/**
+ * Create a data effect
+ * \return Data effect instance
+ */
+RZ_API RzILDataEffect *rz_il_effect_new_data(void) {
+	RzILDataEffect *ret;
+	ret = (RzILDataEffect *)malloc(sizeof(struct rzil_data_effect_t));
+	if (!ret) {
+		return NULL;
+	}
+	ret->operation = 0;
+	ret->var_name = NULL;
+	ret->val = NULL;
+	return ret;
+}
+
+/**
+ * Create a control effect
+ * \return Control effect
+ */
+RZ_API RzILCtrlEffect *rz_il_effect_new_ctrl(void) {
+	RzILCtrlEffect *ret;
+	ret = (RzILCtrlEffect *)malloc(sizeof(struct rzil_control_effect_t));
+	if (!ret) {
+		return NULL;
+	}
+	ret->pc = 0;
+	return ret;
+}
+
+/**
+ * Pack a control effect to a general effect
+ * \param eff control effect
+ * \return general effect
+ */
+RZ_API RzILEffect *rz_il_wrap_ctrl_effect(RzILCtrlEffect *eff) {
+	RzILEffect *ret;
+	ret = (RzILEffect *)malloc(sizeof(struct rzil_effect_union_t));
```

Comment:
Why 2 different types?

Reply:
You mean the `ctrl` one and the `data` one? Because there are 2 different effect types defined in core theory, they may carry different types of info.